src/resource-archive/index.ts

```diff
@@ -128,17 +139,16 @@ class Watcher {
 
     this.firstUrl ??= requestUrl;
 
-    const isLocalRequest =
-      requestUrl.protocol === this.firstUrl.protocol &&
-      requestUrl.host === this.firstUrl.host &&
-      requestUrl.port === this.firstUrl.port;
+    const isRequestFromAllowedDomain =
```

Comment:
Nice! Yeah we shouldn't need to care about the protocol.

Reply:
Oh, this does care (the origin includes the protocol).

Reply:
Ohhh you're right. And that's what you meant by your comment above. I think we only need to check `host`. Do you see any reason why we'd care about protocol?

Reply:
We'd definitely care here (to make sure the request truly is same-origin). But it seems we could do without forcing the user to specify the protocol for external domains.